### PR TITLE
Redirect GET /passwords => /passwords/new

### PR DESCRIPTION
**Pull request by martinemde**
Sometimes you end up on the page `/passwords` and you just want to reload it, but the route doesn't exist.

### CODE REVIEW

File: test/integration/password_reset_test.rb [hunk 1]
```diff
@@ -21,6 +21,12 @@ def forgot_password_with(email)
     perform_enqueued_jobs { click_button "Reset password" }
   end
 
+  test "GET to /passwords redirects to /passwords/new" do
+    visit "/passwords"
+
+    assert_equal new_password_path, page.current_path, "redirects to homepage if link is invalid"
```
Comment (martinemde) [inline test/integration/password_reset_test.rb:27]:
```suggestion
    assert_equal new_password_path, page.current_path, "redirects to new password reset when reloading password create resulting page."
```

**Review by segiddins — APPROVED** [review]

**Comment by simi** [comment]
@martinemde any details on this? :thinking: Ending up on create page with rendered content (usually failed validation is expected and reload should send POST again per my experience. Any idea how to end up on /passwords page to reproduce this problem?

**Comment by martinemde** [comment]
> Ending up on create page with rendered content (usually failed validation is expected and reload should send POST again per my experience. Any idea how to end up on /passwords page to reproduce this problem?

What happens for me sometimes is that I'm on a page with a form that has an expired CSRF token and so submit fails, then I hit cmd+L enter and it fails to load.

**Comment by simi** [comment]
> What happens for me sometimes is that I'm on a page with a form that has an expired CSRF token and so submit fails, then I hit cmd+L enter and it fails to load.

@martinemde and you see following?

image

Since that's probably problem for any form submission in the app. :thinking: It would maybe make more sense to redirect to some generic page explaining on error providing next actions like homepage with appropriate flash message.

**Comment by martinemde** [comment]
@simi that is what happens, but my point is that you can't then fix it by loading the page you're on. It's not a valid url. Redirecting to an error page is an anti-pattern imho, since it makes it even harder to go back and try again. Best response here would be to render `/passwords/new` again with a message that says "please try that again". This PR is just a much simpler version of that fix.

**Comment by martinemde** [comment]
Soapbox time (😁): I'm overall unhappy with the way pages in rails work by posting to a page that prevents you from getting back to the url you're at. I've had countless times where I try to release a page after a form submission failure and the url you're on is not valid to GET, so you're kind of stuck. IMHO, hitting enter on the url bar of a page that loaded successfully once should never 404 on fetching again.

**Comment by simi** [comment]
>  This PR is just a much simpler version of that fix.

@martinemde I don't block merging this, feel free to. I'm just afraid over time how we collect more errors we will need to do this for every create/update action (or for every CSRF form in general). :fearful: I'm looking for something to tackle this in general.

>  I've had countless times where I try to reload a page after a form submission failure and the url you're on is not valid to GET, so you're kind of stuck.

I do second this, having page opened in background for some time and trying to submit form is not so uncommon on my side as well. Any idea how other big Rails apps do behave? Do they use Rails default? I checked on GitHub.com and it responds with following (reload will just repeat the same page).

image

**Comment by martinemde** [comment]
Needs to be updated in light of #4657.

**Pull request closed without merging**